# Web Scraping WikiPedia



# 1) Loading the required table from Wikipedia

In [2]:
#importing Required Packages

import pandas as pd

In [3]:
tables=pd.read_html('https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M')

In [4]:
len(tables)

3

In [7]:
df=tables[0]

In [8]:
df.head()

Postal Code           Borough               Neighborhood
0         M1A      Not assigned               Not assigned
1         M2A      Not assigned               Not assigned
2         M3A        North York                  Parkwoods
3         M4A        North York           Victoria Village
4         M5A  Downtown Toronto  Regent Park, Harbourfront

In [9]:
df.shape

(180, 3)

# 2) Removing "Not assigned" Borough

In [12]:
#Removing Not Assigned Borough
df=df[df['Borough']!= "Not assigned"]

In [13]:
df.head()

Postal Code           Borough                                 Neighborhood
2         M3A        North York                                    Parkwoods
3         M4A        North York                             Victoria Village
4         M5A  Downtown Toronto                    Regent Park, Harbourfront
5         M6A        North York             Lawrence Manor, Lawrence Heights
6         M7A  Downtown Toronto  Queen's Park, Ontario Provincial Government

# 3) Checking for Postal Codes that have more than 1 rows

In [17]:
df.describe()

Postal Code     Borough Neighborhood
count          103         103          103
unique         103          10           99
top            M2M  North York    Downsview
freq             1          24            4

We see that all Postal codes have a unique values. Hence we don't need to combine the rows with same postal codes but different neighborhoods.

# 4) Checking for Neighborhoods which have "Not assigned" value but has a value for Borough

In [15]:
df[df['Neighborhood']== "Not assigned"]

Empty DataFrame
Columns: [Postal Code, Borough, Neighborhood]
Index: []

We find that no such values are present when checking for Neighborhoods which have "Not assigned" value but has a value for Borough.  
So we don't need to assign the same neighborhood name as borough name. 

# 5) Checking No of rows in the dataframe

In [22]:
print("There are {} no of rows present".format(df.shape[0]))

There are 103 no of rows present


#### Hence we notice that there are 103 rows present.